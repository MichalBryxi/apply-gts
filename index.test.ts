import { apply, diff, diffSummary, newEmberApp } from 'ember-apply/test-utils';
import { describe, expect, it } from 'vitest';

import { default as applyGts } from './index.js';

describe('apply-gts', () => {
  it('default export exists', () => {
    expect(typeof applyGts).toEqual('function');
  });

  describe('applying to an ember app', () => {
    it('works via CLI', async () => {
      let appLocation = await newEmberApp();

      await apply(appLocation, applyGts.path);

      expect(await diffSummary(appLocation)).toMatchSnapshot();
      expect(
        await diff(appLocation, { ignoreVersions: true })
      ).toMatchSnapshot();
    });
  });
});
