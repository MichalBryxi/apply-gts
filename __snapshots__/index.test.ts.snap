// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`apply-gts > applying to an ember app > works via CLI 1`] = `
" .prettierrc.js | 6 ++++--
 package.json   | 4 ++++
 2 files changed, 8 insertions(+), 2 deletions(-)"
`;

exports[`apply-gts > applying to an ember app > works via CLI 2`] = `
"diff --git a/.prettierrc.js b/.prettierrc.js
index e5f7b6d..f34fa08 100644
--- a/.prettierrc.js
+++ b/.prettierrc.js
@@ -1,12 +1,14 @@
 'use strict';
 
 module.exports = {
+  plugins: ["prettier-plugin-ember-template-tag"],
+
   overrides: [
     {
-      files: '*.{js,ts}',
+      files: "*.{js,ts,gjs,gts}",
       options: {
         singleQuote: true,
       },
     },
-  ],
+  ]
 };

package.json
{
  "name": "test-app",
  "version": "0.0.0",
  "private": true,
  "description": "Small description for test-app goes here",
  "repository": "",
  "license": "MIT",
  "author": "",
  "directories": {
    "doc": "doc",
    "test": "tests"
  },
  "scripts": {
    "build": "ember build --environment=production",
    "lint": "concurrently \\"npm:lint:*(!fix)\\" --names \\"lint:\\"",
    "lint:css": "stylelint \\"**/*.css\\"",
    "lint:css:fix": "concurrently \\"npm:lint:css -- --fix\\"",
    "lint:fix": "concurrently \\"npm:lint:*:fix\\" --names \\"fix:\\"",
    "lint:hbs": "ember-template-lint .",
    "lint:hbs:fix": "ember-template-lint . --fix",
    "lint:js": "eslint . --cache",
    "lint:js:fix": "eslint . --fix",
    "start": "ember serve",
    "test": "concurrently \\"npm:lint\\" \\"npm:test:*\\" --names \\"lint,test:\\"",
    "test:ember": "ember test"
  },
  "devDependencies": [
    "@babel/core",
    "@babel/eslint-parser",
    "@babel/plugin-proposal-decorators",
    "@ember/optional-features",
    "@ember/string",
    "@ember/test-helpers",
    "@glimmer/component",
    "@glimmer/tracking",
    "@glint/core",
    "@glint/environment-ember-template-imports",
    "broccoli-asset-rev",
    "concurrently",
    "ember-auto-import",
    "ember-cli",
    "ember-cli-app-version",
    "ember-cli-babel",
    "ember-cli-clean-css",
    "ember-cli-dependency-checker",
    "ember-cli-htmlbars",
    "ember-cli-inject-live-reload",
    "ember-cli-sri",
    "ember-cli-terser",
    "ember-data",
    "ember-fetch",
    "ember-load-initializers",
    "ember-modifier",
    "ember-page-title",
    "ember-qunit",
    "ember-resolver",
    "ember-source",
    "ember-template-imports",
    "ember-template-lint",
    "ember-welcome-page",
    "eslint",
    "eslint-config-prettier",
    "eslint-plugin-ember",
    "eslint-plugin-n",
    "eslint-plugin-prettier",
    "eslint-plugin-qunit",
    "loader.js",
    "prettier",
    "prettier-plugin-ember-template-tag",
    "qunit",
    "qunit-dom",
    "stylelint",
    "stylelint-config-standard",
    "stylelint-prettier",
    "tracked-built-ins",
    "webpack"
  ],
  "engines": [
    "node"
  ],
  "ember": {
    "edition": "octane"
  }
}
"
`;
